advanced-usage/mocking-non-public-members-and-types.md

```diff
@@ -998,118 +1075,65 @@ With the next sample, we can handle an even more complex scenario - mock interna
     End Sub
   {{endregion}}
 
-The sample test verifies whether the call to `Create` method has been made and the returned `Web.WebRequest` object has an expected value for `RequestUri` property set.
+##  Mocking Using a Dynamic Wrapper
 
-## Mock Protected Member
+JustMock leverages .NET 4 and the [DLR](https://en.wikipedia.org/wiki/Dynamic_Language_Runtime) to allow you to wrap non-public implementations in `dynamic` objects and arrange them as naturally as you can do it with public members.
 
-To show how to mock a protected member, we will use the following class:
+**Example 16** shows how to wrap the mock instance in a dynamic object using `Mock.NonPublic.Wrap()`. The wrapper can be passed to `Mock.NonPublic.Arrange` and `Mock.NonPublic.Assert` together with an operation to specify what you want to arrange. You could also arrange:
+- the value of a property getter 
+- the action of a property setter
 
-  #### __[C#]__
+Argument matchers are specified using `Arg.Expr`.
 
-  {{region NonPublicMocking#PublicClassWithProtectedMembers}}
-    public class FooWithProtectedMembers
-    {
-        protected virtual void Load()
-        {
-            throw new NotImplementedException();
-        }
+When using dynamic expressions in Visual C# projects, you should reference the Microsoft.CSharp assembly.
```

Comment:
Can we put a link to the nugget? https://www.nuget.org/packages/Microsoft.CSharp/